mmdet/datasets/openimages.py

```diff
@@ -0,0 +1,845 @@
+                    Labels that are human-verified to be absent from an image
+                    have confidence = 0 (negative labels). Machine-generated
+                    labels have fractional confidences, generally >= 0.5.
+                    The higher the confidence, the smaller the chance for
+                    the label to be a false positive.
+        """
+
+        item_lists = defaultdict(list)
+        with open(image_level_ann_file, 'r') as f:
+            reader = csv.reader(f)
+            i = -1
+            for line in reader:
+                i += 1
+                if i == 0:
+                    continue
+                else:
+                    img_id = line[0]
+                    item_lists[img_id].append(
+                        dict(
+                            image_level_label=int(self.index_dict[line[2]]),
+                            confidence=float(line[3])))
+        return item_lists
+
+    def get_image_level_ann(self, image_level_ann_file):
+        """Get OpenImages annotation by index.
+
+        Args:
+            image_level_ann_file (str): CSV style image level annotation
+                file path.
+
+        Returns:
+            dict: Annotation info of specified index.
+        """
+
+        item_lists = self.load_image_label_from_csv(image_level_ann_file)
+        image_level_annotations = []
+        for i in range(len(self)):
+            img_info = self.data_infos[i].get('img_info', None)
+            if img_info is not None:
+                # for Open Images Challenges
+                img_id = img_info['filename'].split('/')[-1][:-4]
+            else:
+                # for Open Images v6
+                img_id = self.data_infos[i]['img_id']
+            item_list = item_lists.get(img_id, None)
+            if item_list is not None:
+                image_level_labels = []
+                confidences = []
+                for obj in item_list:
+                    image_level_label = int(obj['image_level_label'])
+                    confidence = float(obj['confidence'])
+
+                    image_level_labels.append(image_level_label)
+                    confidences.append(confidence)
+
+                if not image_level_labels:
+                    image_level_labels = np.zeros((0, ))
+                    confidences = np.zeros((0, ))
+                else:
+                    image_level_labels = np.array(image_level_labels)
+                    confidences = np.array(confidences)
+            else:
+                image_level_labels = np.zeros((0, ))
+                confidences = np.zeros((0, ))
+            ann = dict(
+                image_level_labels=image_level_labels.astype(np.int64),
+                confidences=confidences.astype(np.float32))
+            image_level_annotations.append(ann)
+
+        return image_level_annotations
+
+    def denormalize_gt_bboxes(self, annotations):
```

Comment:
It also shows up in loading pipeline. Considering to only keep one of them to unify the usages with efficiency

Reply:
Maybe we should keep these two functions, because, during training, we denormalize the gt_bbox in `LoadAnnotations`, which will not be used during testing. Meanwhile, if we want to denormalize gt boxes in OpenImages Dataset, we should get their metafiles first.

Reply:
OK understand...